src/service/resources-impl.js

```diff
@@ -387,15 +388,40 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
```

Comment:
It'd be good. The only question I'd ask: the current approach is nice because it builds elements in the order they are registered. I don't think the order matters, but let's double-check if we may have some negatives.